From 820f3be27faebb0ccfeadc6cf18346cbaa3ba42a Mon Sep 17 00:00:00 2001
From: Gregory CLEMENT <gregory.clement@free-electrons.com>
Date: Wed, 14 Dec 2016 17:43:48 +0100
Subject: [PATCH 1176/1345] ARM64: dts: marvell: armada37xx: add pinctrl
 definition

Start to populate the device tree of the Armada 37xx with the pincontrol
configuration used on the board providing a dts.

Cherry-picked from branch linux-4.12-devel and minor changes because of
the slightly different dts version.

Change-Id: Ib3e10c259b28f3a8c34cc9f1c3f27edab7e8f4de
Signed-off-by: Gregory CLEMENT <gregory.clement@free-electrons.com>
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42773
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Evan Wang <xswang@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts |    2 +-
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   |    9 ---------
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index d9f94a4..81756eb 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -239,7 +239,7 @@
 
 &uart1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_pins>;
+	pinctrl-0 = <&uart2_pins>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index d45b7d0..82ec974 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -349,15 +349,6 @@
 					marvell,pins = "mpp36";
 					marvell,function = "spi";
 				};
-
-				spi_quad_pins: spi-quad-pins {
-					marvell,pins = "mpp15", "mpp16";
-					marvell,function = "spi-quad";
-				};
-				uart1_pins: uart1_pins {
-					marvell,pins = "mpp18", "mpp19";
-					marvell,function = "uart2";
-				};
 			};
 
 			cpu_pm_clk: cpu-pm-clk@14018 {
-- 
1.7.9.5

